### PR TITLE
Add optimistic lock in dref

### PR DESCRIPTION
**Pull request by k9845**
Addresses xxxxxx 

## Changes

* Detailed list or prose of changes
* ...

## Checklist 
Things that should succeed before merging.

- [ ] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

File: dref/serializers.py [hunk 1]
```diff
@@ -348,6 +348,7 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
         validated_data['modified_by'] = self.context['request'].user
         is_assessment_report = validated_data.get('is_assessment_report')
+        modified_at = validated_data.get('modified_at')
```
Comment (frozenhelium) [inline dref/serializers.py:351]: Use pop, use None as default value

File: dref/serializers.py [hunk 2]
```diff
@@ -373,18 +374,21 @@ def update(self, instance, validated_data):
...
-                lambda: send_dref_email.delay(dref.id, list(to), 'Updated')
-            )
-        return dref
+        if modified_at and instance.modified_at and modified_at > instance.modified_at:
+            dref = super().update(instance, validated_data)
+            if to:
+                transaction.on_commit(
+                    lambda: send_dref_email.delay(dref.id, list(to), 'Updated')
+                )
+            return dref
```
Comment (frozenhelium) [inline dref/serializers.py:390]: Throw error if
- modified_at is None, 'modified_at is required'
- modified_at != instance.modified_at, 'OBSOLETE_PAYLOAD'
Note: defined 'OBSOLETE_PAYLOAD' as a static value, client will use this string to identify this type of error. Please add a note in code for this as well.

File: dref/test_views.py [hunk 3]
```diff
@@ -847,3 +848,27 @@ def test_dref_for_super_user(self):
...
+        )
+        url = f'/api/v2/dref/{dref.id}/'
+        data = {
+            'title': "New title",
+            'modified_at': datetime(2022, 2, 18, 2, 29, 39, 793615),
+        }
+
+        self.client.force_authenticate(self.user)
+        response = self.client.patch(url, data=data)
+        self.assertEqual(response.status_code, 200)
```
Comment (frozenhelium) [inline dref/test_views.py:866]: This should be 400, also assert for the error response
Add a test case to verify `modified_at` is not null

**Review by frozenhelium — CHANGES_REQUESTED** [review]

File: dref/models.py [hunk 4]
```diff
@@ -227,7 +228,10 @@ class Status(models.IntegerChoices):
         COMPLETED = 1, _('Completed')
 
     created_at = models.DateTimeField(verbose_name=_('created at'), auto_now_add=True)
-    modified_at = models.DateTimeField(verbose_name=_('modified at'), auto_now=True)
+    modified_at = models.DateTimeField(
+        verbose_name=_('modified at'),
+        default=datetime.now, blank=True
```
Comment (thenav56) [inline dref/models.py:233]: Use auto_now instead of datetime.now
https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.DateField.auto_now
